arch/arm/core/cortex_m/pm_s2ram.S

```diff
@@ -67,10 +114,16 @@ SECTION_FUNC(TEXT, arch_pm_s2ram_suspend)
 	mrs	r2, control
 	str	r2, [r1, #___cpu_context_t_control_OFFSET]
 
+	/*
+	 * Save the return address from `pm_s2ram_mark_set`
+	 */
+	mov r1, r15
+	add r1, #6
```

Comment:
Can you add comment why 6 is added? is this 3 16 bit instructions (mov add b)? `B` can only jump +-2kB. Maybe it would be better to store LR in R1 and use branch with link?

Reply:
> Maybe it would be better to store LR in R1 and use branch with link?
Sounds reasonable - done